### DISCUSSION TITLE
Metrics update period in Linkerd proxy

### DISCUSSION THREAD

**Discussion by nfsp3k** [linkerd/linkerd2#4477] [General] [answered]
Hi,
I'm trying to change the period of metrics update in the Linkerd proxy.

I collected some HTTP metrics (e.g., requests per second, success rate) every second from the Linkerd proxy by using custom metrics APIs that I define in the Prometheus adapter configuration.
But, I found that the HTTP metrics are updated every 5-10 seconds.

How can I change the update period for HTTP metrics provided by the Linkerd proxy?
I want to reduce the update period of the metrics and increase the metrics resolution.
Are there any config flags for which I can use?

Thank you in advance!

**Comment by grampelberg** [comment, marked as answer]
Metrics are scraped every 10s by prometheus. You can modify the scrape config and change it to whatever you'd like. The tradeoff is that you'll be increasing the cpu/memory usage of prometheus and increasing the load on all your proxies by a small amount.

**Reply by nfsp3k** [reply]
Thank you very much!!
I changed the field 'scrape_interval' from 10s to 1s in the ConfigMap resource and I found it works correctly :D
